internal/common/entitlement/connectors.go

```diff
@@ -0,0 +1,96 @@
+package entitlement
+
+import (
+	"log/slog"
+	"time"
+
+	"github.com/openmeterio/openmeter/internal/ent/db"
+	"github.com/openmeterio/openmeter/internal/meter"
+	"github.com/openmeterio/openmeter/openmeter/credit"
+	creditpgadapter "github.com/openmeterio/openmeter/openmeter/credit/postgresadapter"
+	"github.com/openmeterio/openmeter/openmeter/entitlement"
+	booleanentitlement "github.com/openmeterio/openmeter/openmeter/entitlement/boolean"
+	meteredentitlement "github.com/openmeterio/openmeter/openmeter/entitlement/metered"
+	entitlementpgadapter "github.com/openmeterio/openmeter/openmeter/entitlement/postgresadapter"
+	staticentitlement "github.com/openmeterio/openmeter/openmeter/entitlement/static"
+	"github.com/openmeterio/openmeter/openmeter/event/publisher"
+	"github.com/openmeterio/openmeter/openmeter/productcatalog"
+	productcatalogpgadapter "github.com/openmeterio/openmeter/openmeter/productcatalog/postgresadapter"
+	"github.com/openmeterio/openmeter/openmeter/streaming"
+)
+
+type Connectors struct {
+	Feature            productcatalog.FeatureConnector
+	EntitlementOwner   credit.OwnerConnector
+	CreditBalance      credit.BalanceConnector
+	Grant              credit.GrantConnector
+	MeteredEntitlement meteredentitlement.Connector
+	Entitlement        entitlement.EntitlementConnector
+}
+
+type ConnectorsOptions struct {
+	DatabaseClient     *db.Client
+	StreamingConnector streaming.Connector
+	Logger             *slog.Logger
+	MeterRepository    meter.Repository
+	Publisher          publisher.TopicPublisher
+}
+
+func GetConnectors(opts ConnectorsOptions) *Connectors {
```

Comment:
i don't think `internal/common/entitlement` is warranted. If you want to use it for server setup it could go to `internal/server` somewhere, if its needed each time entitlements is used it could go into `internal/entitlement`... I think "common" is a bit like util, it easily gets out of hand. Wdyt?

Reply:
It's for component setups (not just server, but balance-worker, notifications worker); I didn't want to put this into `internal/entitlement` as it also provides product catalog + credit. But I am completely fine putting it there into a registry package or something

Reply:
On the secound tought I would rather have an `internal/registry` package that can yield these connector structs, such as Entitlements (that includes credit + product catalog) or the streaming connector.

Reply:
I generally prefer keeping wiring in main, but if it's too big a burden we can think about introducing some alternative dependency injection strategy.

I've been experimenting with this a while ago: https://github.com/sagikazarmark/go-service-locator

There is Google's wire too: https://github.com/google/wire

Uber has fx: https://github.com/uber-go/fx

I'd like to keep it as simple and as framework independent as possible though.

I'd propose going with something simple for now and then sitting down and talking through some long-term options in the coming weeks.

Reply:
Ok, for now I have created a repository package for this one only. If we end up having more such components where you need to always initialize, we can put additional stuff there.

It's included in this [PR](https://github.com/openmeterio/openmeter/pull/1278), I will close this (I didn't wanted to force push this one with the new contents).